### PR TITLE
Archiver scenario and SQL scripts

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
 - executing archiver scenario

**Related issue(s)**

### CODE REVIEW

File: utils/archiver/db_scripts.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+select count(*) from runtime_states;
+```
+
+## Statements to verify the archiver work
+1. Number of all operations:
+```sql
+select count(*) from operations;
+```
+
+2. Number of operations, which will be deleted by the archiver (belongs to deprovisioned instances):
```
Comment (jaroslaw-pieszka) [inline utils/archiver/db_scripts.md:26]:
```suggestion
2. Number of operations, which will be deleted by the archiver (belonging to deprovisioned instances):
```

File: utils/archiver/db_scripts.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# SQL statements to verify the archiver work and check DB about deprovisioned instances
+
+## DB statistics which should be run before the archiver is started
+
+1. Number of all operations:
```
Comment (IwonaLanger) [inline utils/archiver/db_scripts.md:5]: If the sequence is not important here, please change the ordered lists to unordered ones. If it matters, just ignore this comment.

File: utils/archiver/scenario.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Running Archiver scenario
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:1]:
```suggestion
# Running the Archiver Job Scenario
```

File: utils/archiver/scenario.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+## Gather DB statistics
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:3]:
```suggestion
## Gather DB Statistics
```

File: utils/archiver/scenario.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+Run all SQL statements defined in [DB statistics which should be run before the archiver is started](db_scripts.md#db-statistics-which-should-be-run-before-the-archiver-is-started) section.
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:5]:
```suggestion
Run all the SQL statements defined in the [DB statistics which should be run before the archiver is started](db_scripts.md#db-statistics-which-should-be-run-before-the-archiver-is-started) section.
```

File: utils/archiver/scenario.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+## Run the archiver with disabled deletion
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:7]:
```suggestion
## Run the Archiver Job with Disabled Deletion
```

File: utils/archiver/scenario.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+Set the following environment variables:
+1. `APP_PERFORM_DELETION` set to `false`.
+2. `APP_DRY_RUN`  set to `false`.
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:11]:
```suggestion
- **APP_PERFORM_DELETION** set to `false`
- **APP_DRY_RUN**  set to `false`
```

File: utils/archiver/scenario.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+Run the archiver (`./apply.sh`).
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:13]: These (lines 9 and 13) seem to be subsequent steps, so I suggest numbering them (while the variables can be turned into bullet points).

File: utils/archiver/scenario.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+## Verify the archiver work
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:15]:
```suggestion
## Verify the Archiver Job's Work
```

File: utils/archiver/scenario.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+Run all SQL statements defined in [Statements to verify the archiver work](db_scripts.md#statements-to-verify-the-archiver-work) section.
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:17]:
```suggestion
Run all the SQL statements defined in the [Statements to verify the archiver work](db_scripts.md#statements-to-verify-the-archiver-work) section.
```

File: utils/archiver/scenario.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+## Delete all instances_archived
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:19]:
```suggestion
## Delete All instances_archived
```

File: utils/archiver/scenario.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
+Delete all rows from instances_archived table. It will be recreated once again. Make sure that the number of operations were not decreased (deletion was not performed)
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:21]:
```suggestion
Delete all rows from the instances_archived table. When it is recreated, make sure that the number of operations is not decreased (deletion was not performed).
```

File: utils/archiver/scenario.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+```sql
+delete from instances_archived;
+```
+
+## Enable archiving and deletion of operations and instances at the and of deprovisioning.
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:26]:
```suggestion
## Enable Archiving and Deletion of Operations and Instances at the End of Deprovisioning
```

File: utils/archiver/scenario.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+```
+archiving:
+    enabled: true
+    dryRun: false
+cleaning:
+    enabled: true
+    dryRun: false
+```
+
+and wait for KEB restart with new configuration.
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:38]:
```suggestion
Wait for KEB to restart with the new configuration.
```

Comment (IwonaLanger) [inline utils/archiver/scenario.md:38]: I recommend numbering these two steps (lines 28 and 38).

File: utils/archiver/scenario.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+## Run the archiver with enabled deletion
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:40]:
```suggestion
## Run the Archiver Job with Enabled Deletion
```

File: utils/archiver/scenario.md [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+
+Run the archiver (`./apply.sh`).
+
+## Verify the Archiver Job's Work
+
+Run all the SQL statements defined in the [Statements to verify the archiver work](db_scripts.md#statements-to-verify-the-archiver-work) section.
+
+## Delete All Archived Instances
+
+Make sure that the number of operations were is not decreased (deletion was not performed).
```
Comment (IwonaLanger) [inline utils/archiver/scenario.md:21]:
```suggestion
Make sure that the number of operations is not decreased (deletion was not performed).
```

File: docs/contributor/06-60-archiver-job.md [hunk 17]
```diff
@@ -1,6 +1,6 @@
 # Archiver Job
 
-The archiver job is a tool for archiving and cleaning the data about already deprovisioned instances. The archiver job is run once. All data about deprovisioned instances in the future will be archived and cleaned by proper deprovisioning steps.
+The archiver job is a tool for archiving and cleaning the data about already deprovisioned instances. The archiver job is run once. All data about deprovisioned instances in the future will be archived and cleaned by proper deprovisioning steps. The scenario of running the job is describe in the [scenario.md](./../../utils/archiver/scenario.md) file.
```
Comment (IwonaLanger) [inline docs/contributor/06-60-archiver-job.md:3]:
```suggestion
The archiver job is a tool for archiving and cleaning the data about already deprovisioned instances. The archiver job is run once. All data about deprovisioned instances in the future will be archived and cleaned by proper deprovisioning steps. The scenario of running the job is described in the [scenario.md](./../../utils/archiver/scenario.md) file.
```

**Review by IwonaLanger — APPROVED** [review]

File: utils/archiver/db_scripts.md [hunk 18]
```diff
@@ -0,0 +1,39 @@
+# SQL Statements to Verify the Archiver Job's Work and Check DB for Deprovisioned Instances
+
+## DB Statistics to Be Run Before Starting the Archiver Job
+
+Number of all operations:
+```sql
+select count(*) from operations;
+```
+
+Number of operations that the archiver job will delete (belongs to deprovisioned instances):
```
Comment (jaroslaw-pieszka) [inline utils/archiver/db_scripts.md:10]:
```suggestion
Number of operations that the archiver job will delete (belonging to deprovisioned instances):
```

**Review by jaroslaw-pieszka — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Comment by piotrmiskiewicz** [comment]
no longer needed

**Pull request closed without merging**